modules/npmscripts.config.js

```diff
@@ -797,6 +801,7 @@ module.exports = {
 			'babel-eslint',
 			'browser-tabs-lock',
 			'ckeditor4',
+			'ckeditor5',
```

Comment:
@bryceosterhaus What is the right way to allow third party dependencies? Adding them to this list does not seem to work.

 Curiously, CI SF complains about this, but local `gradle format` and `ant format-source-current-branch` do not.

Reply:
You can run Sf locally with `yarn run node-scripts check:ci` in `./modules` to get the same result I think.

> dependency not provided by a specific module:

This resolve this, I think you need to add the dependencies under `build.imports['frontend-editor-ckeditor-web']` in `./modules/npmscripts.config.js`

Reply:
That build.imports list just tells portal "what module is providing this dependency" meaning it will live under `/o/frontend-editor-ckditor-web`

Reply:
@bryceosterhaus I was able to fix the issue, by importing like you said. Also, `check:ci` works great  :sparkles:

Another issue popped up, I got `Cannot infer exported symbols for ckeditor5: Error [ERR_REQUIRE_ESM]: require() of ES Module`. I was able to fix that one by adding `'ckeditor5': [],` to `modules/node-scripts.config.js` (see last commit).

After that, third issue popped up, and I'm stuck on this one. See failure in https://github.com/liferay-frontend/liferay-portal/pull/4544#issuecomment-2464787270, a bunch of `TS1005: ',' expected` errors. My best guess is that CKEditor 5 was [built with TypeScript 5](https://ckeditor.com/docs/ckeditor5/latest/getting-started/setup/typescript-support.html), and we are [using TypeScript 4 in node scripts](https://github.com/liferay/liferay-portal/blob/master/modules/_node-scripts/package.json#L46). Is this the right diagnosis? If so, is there some place where I can say to SF "don't check this third party module"? Or would it make sense to update TS in DXP?

Reply:
cc @izaera for the `Cannot infer exported symbols` part.

@markocikos Let me pull down your PR and see the TS errors and will let you know

Reply:
@markocikos so the type issue is because I think ckeditor 5 requires TS 4.5+, which allows for the "type" modifier on a named import.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#type-modifiers-on-import-names


We may be able to bump to 4.5... but not sure if that'd raise other issues

Reply:
@markocikos I updated TS here, https://github.com/markocikos/liferay-portal/pull/203

I think this passes now

Reply:
TS has weird versioning, I essentially just jumped versions until one passed 😬

Reply:
Changes in the last push:
- added TS version bump commits from https://github.com/markocikos/liferay-portal/pull/203
- added used `ckeditor5` imports to the list of export symbols 
- added FF
- added a quick & dirty SF fix for compiler error @bryceosterhaus look the other way! :confounded:  This code is very fragile and intentionally has stale state. We can think of refactoring it as a part of CX support in CKE5.

@bryceosterhaus Although SF is now finally green, there is this warning:
```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.7.4
```
Is this safe to ignore?